### DISCUSSION TITLE
Is there any event before onSubmit

### DISCUSSION THREAD

**Discussion by tkatariya** [react-hook-form/react-hook-form#2252] [Q&A]
Hey there, I am trying to implement multi-step form. In this there is submit button on last step. 
Now because there are validations in step 1, form does not gets submitted. What i want is, to jump to step 1 as there are validations in step 1. I want to know what fields have errors. I can use errors from useForm, but there is a lot of logic to handle that and just want to know if there already have an existing api or event.

Here is sample codesandbox  - https://codesandbox.io/s/react-hook-form-nested-fields-3l0wp?file=/src/index.js

**Comment by tkatariya** [comment]
I know you dont call onSubmit function untill all Validations are met, but do i also have that liberty to check from some event?

**Reply by bluebill1049** [reply]
you may want to take a look this question at stackoverlfow: https://stackoverflow.com/questions/62869765/how-to-give-validation-in-multi-step-form-using-react
